lib/authentication.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2017, Kinvey, Inc. All rights reserved.
+ *
+ * This software is licensed to you under the Kinvey terms of service located at
+ * http://www.kinvey.com/terms-of-use. By downloading, accessing and/or using this
+ * software, you hereby accept such terms of service  (and any agreement referenced
+ * therein) and agree that you have read, understand and agree to be bound by such
+ * terms of service and are of legal age to agree to such terms with Kinvey.
+ *
+ * This software contains valuable confidential and proprietary information of
+ * KINVEY, INC and is subject to applicable licensing agreements.
+ * Unauthorized reproduction, transmission or distribution of this file and its
+ * contents is a violation of applicable laws.
+ */
+
+const { HTTPMethod, LogLevel } = require('./constants');
+const { Endpoints } = require('./utils');
+
+/**
+ * Handles authentication: set/unset current user, login/logout.
+ */
+class Authentication {
+  constructor(cliManager) {
+    this.cliManager = cliManager;
+    this._currentUser = null;
+  }
+
+  getCurrentUser() {
+    return this._currentUser;
+  }
+
+  setCurrentUser({ email, token, host }) {
+    this._currentUser = {
+      email,
+      token,
+      host
+    };
+  }
+
+  hasCurrentUser() {
+    return this._currentUser && this._currentUser.token;
+  }
+
+  login(email, password, MFAToken, host, done) {
+    const data = {
+      email,
+      password
+    };
+
+    if (MFAToken) {
```

Comment:
`if (thing)` should only be used if `thing` is expected to be `true` or `false` (i.e. `if (thing === true)`). For null/undefined, explicit checks `if (MFAToken != null)` should be used instead. This applies everywhere in the codebase... (e.g. `if (err != null)`

Reply:
According to the style guide explicit comparison is required only for strings and numbers.
https://github.com/Kinvey/style-guides/blob/master/backend/es6.md#comparison--shortcuts

Also, the paragraph for checking errors in callbacks uses `if (err)`
https://github.com/Kinvey/style-guides/blob/master/backend/es6.md#requires--check-errors